testing/adios2/interface/TestADIOSInterface.cpp

```diff
@@ -77,6 +78,259 @@ TEST_F(ADIOS2_CXX11_API_IO, EngineDefault)
+
+        var.SetSelection(myData.selection(b));
+        engine.Put(var, &myData[b][0], adios2::Mode::Deferred);
+    }
+    engine.Close();
+
+    EXPECT_TRUE(checkOutput("multi_deferred.bp"));
+}
+
+TEST_F(ADIOS2_CXX11_API_Put, MultiBlockPutDS)
+{
+    SetupDecomposition(10);
+
+    adios2::Engine engine = io.Open("multi_ds.bp", adios2::Mode::Write);
+    adios2::Variable<T> var = io.DefineVariable<T>("var", m_Shape);
+
+    MyData<T> myData(m_Selections);
+
+    for (int b = 0; b < myData.nBlocks(); ++b)
+    {
+        PopulateBlock(myData, b);
+
+        var.SetSelection(myData.selection(b));
+        if (b == 0)
+        {
+            engine.Put(var, &myData[b][0], adios2::Mode::Deferred);
+        }
+        else
+        {
+            engine.Put(var, &myData[b][0], adios2::Mode::Sync);
+        }
+    }
+    engine.Close();
+
+    EXPECT_TRUE(checkOutput("multi_ds.bp"));
+}
+
+TEST_F(ADIOS2_CXX11_API_Put, MultiBlockPutZeroCopySync)
+{
+    SetupDecomposition(10);
+
+    adios2::Engine engine = io.Open("multi0_sync.bp", adios2::Mode::Write);
+    adios2::Variable<T> var = io.DefineVariable<T>("var", m_Shape);
+
+    MyDataView<T> myData(m_Selections);
+    for (int b = 0; b < myData.nBlocks(); ++b)
+    {
+        var.SetSelection(myData.selection(b));
+        auto span = engine.Put(var);
+        myData.place(b, span.data());
+    }
+
+    for (int b = 0; b < myData.nBlocks(); ++b)
+    {
+        PopulateBlock(myData, b);
+    }
+    engine.Close();
+
+    EXPECT_TRUE(checkOutput("multi0_sync.bp"));
+}
+
+TEST_F(ADIOS2_CXX11_API_Put, MultiBlockPutZeroCopySync2)
+{
+    SetupDecomposition(10);
+
+    adios2::Engine engine = io.Open("multi0_sync2.bp", adios2::Mode::Write);
+    adios2::Variable<T> var = io.DefineVariable<T>("var", m_Shape);
+
+    MyDataView<T> myData(m_Selections);
+    for (int b = 0; b < 1; ++b)
+    {
+        var.SetSelection(myData.selection(b));
+        auto span = engine.Put(var);
+        myData.place(b, span.data());
+    }
+
+    for (int b = 0; b < 1; ++b)
+    {
+        PopulateBlock(myData, b);
+    }
+    std::vector<T> lastBlock(m_Nx / 2);
+    std::iota(lastBlock.begin(), lastBlock.end(), T(rank * m_Nx + m_Nx / 2));
+    var.SetSelection(myData.selection(1));
+    engine.Put(var, lastBlock.data(), adios2::Mode::Deferred);
+    engine.Close();
+
+    EXPECT_TRUE(checkOutput("multi0_sync2.bp"));
+}
+
+#if 0
+TEST_F(ADIOS2_CXX11_API_Put, MultiBlockPutZeroCopySync3)
+{
+    SetupDecomposition(10);
+
+    adios2::Engine engine = io.Open("multi0_sync3.bp", adios2::Mode::Write);
+    adios2::Variable<T> var = io.DefineVariable<T>("var", m_Shape);
+
+    MyDataView<T> myData(m_Selections);
+    for (int b = 0; b < 1; ++b)
+    {
+        var.SetSelection(myData.selection(b));
+        auto span = engine.Put(var);
+        myData.place(b, span.data());
+    }
+
+    for (int b = 0; b < 1; ++b)
+    {
+        PopulateBlock(myData, b);
+    }
+
+    engine.Put(var, std::vector<T>{5., 6., 7., 8., 9.}.data(),
```

Comment:
This is interesting, it's an address vector.data(), to a r-value. We'll have to test this one.